tests/ssr.test.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect } from 'react'
+import { act, screen } from '@testing-library/react'
+import { hydrateRoot } from 'react-dom/client'
+import { renderToString } from 'react-dom/server'
+import { describe, expect, it } from 'vitest'
+import { create } from 'zustand'
+
+const IS_REACT_18 = React.version.startsWith('18')
```

Comment:
We can skip any react version under 18

```suggestion
import React, { useEffect } from 'react'

if (!React.version.startsWith('18')) {
  describe.skip('skipping test on react version < 18', () => {})
  return
}

import { act, screen } from '@testing-library/react'
import { hydrateRoot } from 'react-dom/client'
import { renderToString } from 'react-dom/server'
import { describe, expect, it } from 'vitest'
import { create } from 'zustand'
```

Reply:
Oh, thank you for your review.
I understand your intention.
I resolved it by using the only method from vitest and dynamic import.
What do you think about it?

Reply:
@NaamuKim that approach doesn't work you should completely avoid import react-dom/client. Let's try with my suggestion.

Reply:
There's an issue with top-level returns not being allowed in ESM, causing errors. 
I think your suggestion of using describe.skip is excellent, and I've configured it to conditionally branch describe based on the version. 
Could you please confirm this?

Reply:
Looks good to me